cli/azd/ci-build.ps1

```diff
@@ -1,5 +1,57 @@
 param(
-    [string] $Version,
-    [string] $SourceVersion
+    [string] $Version = (Get-Content "$PSScriptRoot/../version.txt"),
+    [string] $SourceVersion = (git rev-parse HEAD)
 )
-go build -ldflags="-X 'github.com/azure/azure-dev/cli/azd/internal.Version=$Version (commit $SourceVersion)'"
\ No newline at end of file
+
+# On Windows, use the goversioninfo tool to embed the version information into the executable.
+if ($IsWindows) {
+    Write-Host "Windows build, set verison info and run 'go generate'"
+    if (! (Get-Command "goversioninfo" -ErrorAction SilentlyContinue)) {
+        Write-Host "goversioninfo not found, installing"
+        go install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.4.0
+        Get-Command "goversioninfo" -ErrorAction Stop
+    }
```

Comment:
`go generate` should use `GOBIN` by default anyway, so trying to get the command via `PATH` and update `PATH` accordingly seems extraneous. If `go install` fails it'll set `$LASTEXITCODE`, which you could check instead to terminate:

```pwsh
go install ...
if ($LASTEXITCODE) {
  throw "Failed to install go tool <url>"
}
```

Reply:
The install does succeed but we were failing during build because the go bin directory is not placed in `PATH` ([just the location of the go tool](https://github.com/microsoft/azure-pipelines-tasks/blob/482d590ad3a1254a14947635155159d36dfbc42d/Tasks/GoToolV0/gotool.ts#L39)) so tools installed via `go install` weren't reachable [without intervention](https://github.com/Azure/azure-dev/pull/1457/files#diff-130ac4d47b2be4d9e6d5d602b24dc9e2a8dca24d3746911043a1132b84113184).

`Get-Command` is used to ensure that the tool installed successfully _and_ is reachable.

Reply:
Were you running `go generate`? Odd, if so. Go should automatically use `GOBIN` regardless of the `PATH`.

Reply:
Yes. I run `go generate` on line 34